pkg/webhook/ephemeraljob/validating/ephemeraljob_create_update_handler.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2021 The Kruise Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validating
+
+import (
+	"context"
+	"net/http"
+
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	appsv1alpha1 "github.com/openkruise/kruise/apis/apps/v1alpha1"
+)
+
+// EphemeralJobCreateUpdateHandler handles EphemeralJob
+type EphemeralJobCreateUpdateHandler struct {
+	// Decoder decodes objects
+	Decoder *admission.Decoder
+}
+
+var _ admission.Handler = &EphemeralJobCreateUpdateHandler{}
+
+func NewHandler(mgr manager.Manager) admission.Handler {
+	return &EphemeralJobCreateUpdateHandler{Decoder: admission.NewDecoder(mgr.GetScheme())}
+}
+
+// Handle handles admission requests.
+func (h *EphemeralJobCreateUpdateHandler) Handle(ctx context.Context, req admission.Request) admission.Response {
+	obj := &appsv1alpha1.EphemeralJob{}
+
+	err := h.Decoder.Decode(req, obj)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	if err := validate(obj); err != nil {
+		klog.Warningf("Error validate EphemeralJob %s: %v", obj.Name, err)
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	return admission.ValidationResponse(true, "allowed")
+}
+
+func validate(obj *appsv1alpha1.EphemeralJob) error {
+
+	return nil
```

Comment:
include some logics from validateEphemeralContainers in k8s.io/kubernetes/pkg/apis/core/validation/validation.go